src/librustdoc/html/static/js/search.js

```diff
@@ -154,13 +389,101 @@ window.initSearch = function(rawSearchIndex) {
             return NO_TYPE_FILTER;
         }
 
-        var valLower = query.query.toLowerCase(),
-            val = valLower,
-            typeFilter = itemTypeFromName(query.type),
-            results = {}, results_in_args = {}, results_returned = {},
-            split = valLower.split("::");
+        userQuery = userQuery.trim();
+        var query = {
+            original: userQuery,
+            userQuery: userQuery.toLowerCase(),
+            length: userQuery.length,
+            pos: 0,
+            typeFilter: null,
+            elems: [],
+            elemName: null,
+            args: [],
+            returned: [],
+            // Total number of elements (includes generics).
+            totalElems: 0,
+            // Total number of "top" elements (does not include generics).
+            foundElems: 0,
+            // This field is used to check if it's needed to re-run a search or not.
+            id: "",
```

Comment:
This field is set to `userQuery`, and the check to re-run a search checks whether the new query.id is equal to it. Why not skip the indirection and check whether `query.userQuery == currentResults`? Then we could get rid of this field.

Reply:
Good idea!